src/functional/fencil_processors/codegens/gtfn/itir_to_gtfn_ir.py

```diff
@@ -172,9 +172,9 @@ def _collect_offset_definitions(
     return offset_definitions
 
 
+@dataclasses.dataclass(frozen=True)
 class GTFN_lowering(eve.NodeTranslator, eve.VisitorWithSymbolTableTrait):
```

Comment:
Except for the `uids` change, the changes here are not required, but a partial cleanup of the pass such that the pass is still a frozen dataclass.